src/main/python/hydra/lib/hydrabase.py

```diff
@@ -111,8 +119,48 @@ def delete_tasks_from_group(self, num_tasks_to_delete):
                 l.info("Error in deletiong task: %s " % str(e))
                 return 1
         self.hydra.refresh_app_info(self.app_name)
+
+        # Make sure that desired number of tasks have been deleted.
+        num_tasks_after_delete = self.get_num_tasks()
+        if current_num_tasks_in_grp - num_tasks_to_delete != num_tasks_after_delete:
+            l.error("Error in deleting tasks from group")
+            return 1
         return deleted_tasks
 
+    def wait_for_plan_done(self, timeout=30, sleep_before_next_try=0.2):
```

Comment:
this function should not be here as in hydra does not know what plan is being run by the underlying application as such it has no way of querying the plan status.
This assumes that every app will have uniform get_plan_status method. 
Its the job of the underlying distributed app  to implement and use the get_plan_status methods.
If we really really want to create methods in hydra to query plan status for underlying apps, hydra should make sure that it explicitly asks for the get_plan_status method to be provided as argument by the caller and should assert if the method is not callable.